torch/csrc/distributed/rpc/rpc_command_base.h

```diff
@@ -0,0 +1,21 @@
+#pragma once
+
+#include <torch/csrc/distributed/rpc/message.h>
+
+namespace torch {
+namespace distributed {
+namespace rpc {
+
+// Base class for all RPC request and responses.
+class RpcCommandBase {
+ public:
+  // Need to override this to serialize the RPC.
+  virtual Message toMessage() = 0;
```

Comment:
According to the documentation for `RpcWithAutograd`, this is destructive. Is that true for all classes? In that case, it should be documented here. Furthermore, you can make this take `this` by move by defining this as:

```
virtual Message toMessage() && = 0;
```

and then changing call sites to:

```
auto msg = std::move(rpc_command).toMessage();
```

which will make it more obvious to end users that the destruction is happening.

Reply:
That... seems very questionable from an API design perspective?